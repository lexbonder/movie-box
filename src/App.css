@import url('https://fonts.googleapis.com/css?family=Russo+One');

* {
  box-sizing: border-box;
}

body {
  background-color: rgba(0, 0, 0, 0.885);
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

header {
  background-image: url('./images/popcorn.jpg');
  background-size: 690px;
  height: 200px;
  margin-bottom: 10px;
  opacity: 0.6;
  padding: 20px;
  z-index: -1;
}

.heading {
  color: rgb(255, 255, 255);
  font-family: 'Russo One', sans-serif;
  font-size: 35px;  
  position: absolute;
  display: inline-block;
  top: 0px;
  left: 480px;
  width: 30%;
  z-index: 3;  
}

.App-intro {
  font-size: large;
}

.nav {
  color: rgb(169, 0, 0);
  font-size: 18px;
  margin: 0 0 0px 20px;
  position: relative;
  left: 600px;
  top: -190px;
}

.nav:hover {
  color: rgb(255, 2, 2);
}

.active {
  /* For the NavLink, comes default, changes the current page's tab */
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
