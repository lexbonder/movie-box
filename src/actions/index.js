//Add Favorites
//Toggle Favorites
//Set Filter