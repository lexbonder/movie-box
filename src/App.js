import React, { Component } from 'react';
import fetchApi from './apiCall.js';
import { connect } from 'react-redux';
import { updateMovie } from './actions';
import './App.css';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount = async () => {
    const movies = await fetchApi();
    this.sendMoviesToStore(movies);
  };

  sendMoviesToStore = (movies) => {
    movies.forEach(movie => {
      this.props.addMovie(movie)
    });
  }

  render() {
    console.log(this.props);
    //url root for images = https://image.tmdb.org/t/p/w500/
    return (
      <div className="App">
        <h1>movie tracker beyotch</h1>
        
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  movies: state.movies
});

export const mapDispatchToProps = dispatch => ({
  addMovie: movie => dispatch(updateMovie(movie))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);

