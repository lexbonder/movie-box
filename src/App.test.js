import React from 'react';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { updateMovie } from './actions';
import { shallow } from 'enzyme';

describe('App ', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  // it('calls updateMovie when sendMoviesToStore is called', () => {
  //   const mockUpdateMovie = jest.fn()
  //   const mockMovies = [{movie: 'name'}]
  //   const wrapper = shallow(<App updateMovie={mockUpdateMovie} />);
  //   wrapper.instance().sendMoviesToStore(mockMovies);
  //   expect(mockUpdateMovie).toHaveBeenCalled();
  // })

  it('map the store correctly with MSTP', () => {
    const wrapper = shallow(<App />);
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.changeStore();
    expect(mockDispatch).toHaveBeenCalled();
  });

});