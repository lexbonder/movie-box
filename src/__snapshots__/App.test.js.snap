// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={
      Object {
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo1"
            to="/"
          >
            Movie
          </NavLink>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo2"
            to="/"
          >
            Tracker
          </NavLink>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="login"
            to="/login/"
          >
            Login / Sign up
          </NavLink>
        </header>,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          path="/login/"
          strict={true}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo1"
              to="/"
            >
              Movie
            </NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo2"
              to="/"
            >
              Tracker
            </NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="login"
              to="/login/"
            >
              Login / Sign up
            </NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Movie",
              "className": "logo1",
              "to": "/",
            },
            "ref": null,
            "rendered": "Movie",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Tracker",
              "className": "logo2",
              "to": "/",
            },
            "ref": null,
            "rendered": "Tracker",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Login / Sign up",
              "className": "login",
              "to": "/login/",
            },
            "ref": null,
            "rendered": "Login / Sign up",
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login/",
          "strict": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo1"
              to="/"
            >
              Movie
            </NavLink>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo2"
              to="/"
            >
              Tracker
            </NavLink>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="login"
              to="/login/"
            >
              Login / Sign up
            </NavLink>
          </header>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/login/"
            strict={true}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="logo1"
                to="/"
              >
                Movie
              </NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="logo2"
                to="/"
              >
                Tracker
              </NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login"
                to="/login/"
              >
                Login / Sign up
              </NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Movie",
                "className": "logo1",
                "to": "/",
              },
              "ref": null,
              "rendered": "Movie",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Tracker",
                "className": "logo2",
                "to": "/",
              },
              "ref": null,
              "rendered": "Tracker",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login / Sign up",
                "className": "login",
                "to": "/login/",
              },
              "ref": null,
              "rendered": "Login / Sign up",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login/",
            "strict": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App  should match the snapshot when the user clicks login/signup 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={
      Object {
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo1"
            to="/"
          >
            Movie
          </NavLink>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="logo2"
            to="/"
          >
            Tracker
          </NavLink>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="login"
            to="/login/"
          >
            Login / Sign up
          </NavLink>
        </header>,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          path="/login/"
          strict={true}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo1"
              to="/"
            >
              Movie
            </NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo2"
              to="/"
            >
              Tracker
            </NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="login"
              to="/login/"
            >
              Login / Sign up
            </NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Movie",
              "className": "logo1",
              "to": "/",
            },
            "ref": null,
            "rendered": "Movie",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Tracker",
              "className": "logo2",
              "to": "/",
            },
            "ref": null,
            "rendered": "Tracker",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Login / Sign up",
              "className": "login",
              "to": "/login/",
            },
            "ref": null,
            "rendered": "Login / Sign up",
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/login/",
          "strict": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo1"
              to="/"
            >
              Movie
            </NavLink>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="logo2"
              to="/"
            >
              Tracker
            </NavLink>
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="login"
              to="/login/"
            >
              Login / Sign up
            </NavLink>
          </header>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            path="/login/"
            strict={true}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="logo1"
                to="/"
              >
                Movie
              </NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="logo2"
                to="/"
              >
                Tracker
              </NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="login"
                to="/login/"
              >
                Login / Sign up
              </NavLink>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Movie",
                "className": "logo1",
                "to": "/",
              },
              "ref": null,
              "rendered": "Movie",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Tracker",
                "className": "logo2",
                "to": "/",
              },
              "ref": null,
              "rendered": "Tracker",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Login / Sign up",
                "className": "login",
                "to": "/login/",
              },
              "ref": null,
              "rendered": "Login / Sign up",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/login/",
            "strict": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App  toggleLogin should match the snapshot when a user is logged in 1`] = `
<App
  user={
    Object {
      "id": 1,
      "name": "joe",
    }
  }
/>
`;

exports[`App  toggleLogin should match the snapshot when a user is not logged in 1`] = `
<App
  user={
    Object {
      "name": "",
    }
  }
/>
`;
